Tests/SonarScanner.MSBuild.PreProcessor.Test/PreProcessorTests.cs

```diff
@@ -499,25 +391,87 @@ private static void AssertExpectedServerSetting(AnalysisConfig actualConfig, str
-
     private sealed class TestScope : IDisposable
     {
         public readonly string WorkingDir;
+        public readonly MockObjectFactory Factory;
+        public readonly PreProcessor PreProcessor;
 
         private readonly WorkingDirectoryScope workingDirectory;
+        private readonly TestContext testContext;
 
-        public TestScope(TestContext context)
+        public TestScope(TestContext context, MockObjectFactory factory = null)
         {
+            testContext = context;
             WorkingDir = TestUtils.CreateTestSpecificFolderWithSubPaths(context);
             workingDirectory = new WorkingDirectoryScope(WorkingDir);
+            Factory = factory ?? new MockObjectFactory();
+            PreProcessor = new PreProcessor(Factory, Factory.Logger);
+        }
+
+        public void AssertDirectoriesCreated()
+        {
+            var settings = Factory.ReadSettings();
+            AssertDirectoryExists(settings.AnalysisBaseDirectory);
+            AssertDirectoryExists(settings.SonarConfigDirectory);
+            AssertDirectoryExists(settings.SonarOutputDirectory);
+            // The bootstrapper is responsible for creating the bin directory
+        }
+
+        public async Task<bool> Execute(IEnumerable<string> args = null)
+        {
+            args ??= CreateArgs();
+            return await PreProcessor.Execute(args);
+        }
+
+        public AnalysisConfig AssertAnalysisConfig(int numAnalyzers)
+        {
+            var filePath = Factory.ReadSettings().AnalysisConfigFilePath;
+            Factory.Logger.AssertNoErrorsLogged();
+            Factory.Logger.AssertVerbosity(LoggerVerbosity.Debug);
+
+            AssertConfigFileExists(filePath);
+            var actualConfig = AnalysisConfig.Load(filePath);
+            actualConfig.SonarProjectKey.Should().Be("key", "Unexpected project key");
+            actualConfig.SonarProjectName.Should().Be("name", "Unexpected project name");
+            actualConfig.SonarProjectVersion.Should().Be("1.0", "Unexpected project version");
+            actualConfig.AnalyzersSettings.Should().NotBeNull("Analyzer settings should not be null");
+            actualConfig.AnalyzersSettings.Should().HaveCount(numAnalyzers);
+
+            AssertExpectedLocalSetting(actualConfig, SonarProperties.HostUrl, "http://host");
+            AssertExpectedLocalSetting(actualConfig, "cmd.line1", "cmdline.value.1");
+            AssertExpectedServerSetting(actualConfig, "server.key", "server value 1");
+
+            return actualConfig;
+        }
+
+        public void AssertAnalysisConfigPathInSonarConfigDirectory()
+        {
+            var settings = Factory.ReadSettings();
+            Directory.Exists(settings.SonarConfigDirectory);
+            var actualFileNames = Directory.GetFiles(settings.SonarConfigDirectory).Select(Path.GetFileName);
+            actualFileNames.Should().BeEquivalentTo(Path.GetFileName(settings.AnalysisConfigFilePath));
```

Comment:
That's a cryptic way how to say the same thing
```suggestion
            actualFileNames.Should().BeEquivalentTo("SonarQubeAnalysisConfig.xml");
```

Reply:
You can also get rid of `actualFileNames` merge those two lines